random/main.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	crand "crypto/rand"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+
+	"github.com/blocky/basm-go-sdk"
+)
+
+type Args struct {
+	Sides uint64 `json:"sides"`
+}
+
+//export rollDie
+func rollDie(inputPtr uint64, secretPtr uint64) uint64 {
+	var input Args
+	inputData := basm.ReadFromHost(inputPtr)
+	err := json.Unmarshal(inputData, &input)
+	switch {
+	case err != nil:
+		outErr := fmt.Errorf("could not unmarshal input args: %w", err)
+		return WriteError(outErr)
+	case input.Sides == 0:
+		outErr := fmt.Errorf("die cannot have zero sides")
+		return WriteError(outErr)
+	}
+
+	randBytes := make([]byte, 8)
+	n, err := crand.Read(randBytes)
```

Comment:
Currently, we show that you can generate random numbers in much the same way you would a normal program. Namely, you import a standard library that pulls entropy from the system which it then uses to generate random streams. This works because tinygo provides a `crypto/rand` implementation which pulls entropy from the `wazero` runtime which we have correctly seeded.

If users don't want to (or can't) depend on having a stdlib implementation like `crypo/rand` we could define a host func in delphi that provides random byte streams by pulling from the `newman` entropy provider. I figure that is something we could push off for now and implement in the future if we see a need for it

Reply:
I think it is fine.  Let's wait to see if anybody wants such a feature.